### ISSUE TITLE
[Bug]:   Unable to draw lines

### ISSUE THREAD

**Issue by dogboydog** [DmitriySalnikov/godot_debug_draw_3d#14]
### Godot version

v3.5.2.stable.mono.official [170ba337a]

### DebugDraw3D version

1.0.2

### On which operating systems the error occurs

Windows

### Using which renderers the error occurs

GLES3

### Issue description

Hello, I'm quite interested to try this plugin for my project as it seems really cool. 

In earlier messages you asked to confirm the plugin is installed by searching the help for DebugDraw I believe.  In my project and my test repro for this issue, I can't see that in the help. 

![image](https://github.com/DmitriySalnikov/godot_debug_draw_3d/assets/9920963/94ec053b-0199-445e-8f39-81ec911ab332)

![image](https://github.com/DmitriySalnikov/godot_debug_draw_3d/assets/9920963/26dab8e8-cfdf-42ff-a85c-9c6ef6e0c32f)

I'm also kind of unclear about how to use this for 2D, is it possible? The arguments are Vector3, but not sure if these will display in a 2D view. I would ideally like to draw lines with with global position /world space position  of 2D nodes and have them display over everything. 

### Steps to reproduce

Open this zip project and run the main scene

The text is showing but the lines are not drawing
```cs
		public override void _Process(float delta)
		{
        DebugDrawCS.Freeze3dRender = true;
        DebugDrawCS.ForceUseCameraFromScene = true;
        DebugDrawCS.DebugEnabled = true;
        DebugDrawCS.VisibleInstanceBounds = true;
        DebugDrawCS.DrawLine(new Vector3(0, 0, 0), new Vector3(200,200,0), Colors.Red, 1f);
        DebugDrawCS.DrawLine(new Vector3(0, 0, 3), new Vector3(1,1,3), Colors.Green, 1f);
        DebugDrawCS.DrawLine(new Vector3(0, 0, 8), new Vector3(8,8,8), Colors.Blue, 1f);
        DebugDrawCS.SetText("Test", "4");
        DebugDrawCS.SetText("Test2", "444");
        DebugDrawCS.GeometryRenderLayers = 6;
		}
````

### Minimal reproduction project

[DebugDrawTest.zip](https://github.com/DmitriySalnikov/godot_debug_draw_3d/files/11534563/DebugDrawTest.zip)

**Comment by DmitriySalnikov** [comment]
```cs
DebugDrawCS.Freeze3dRender = true;
```
It does not allow you to change the geometry that has already been drawn. In this case, nothing was drawn... You need to remove `freeze'.
```cs
DebugDrawCS.GeometryRenderLayers = 6;
```
This is not the 6th layer. These are the 2nd and 3rd layers. Because 6 in binary is equal to 0110.
The 6th layer is 32, or `1 << 5` or you can use one of the methods `set_*_mask_bit` (in this case `set_layer_mask_bit`)
![image](https://github.com/DmitriySalnikov/godot_debug_draw_3d/assets/7782218/4c63a713-5ee4-4305-871e-217d614f91db)
Also, in order to see 3D lines, you need at least one 3D camera on the scene. You can even use this tool:
![image](https://github.com/DmitriySalnikov/godot_debug_draw_3d/assets/7782218/6ce591b7-cfe4-4dcf-a80b-ef08fca2b215)

I was thinking about adding support for 2d debug. Because at the moment, in order to draw a line, you need to manually save it somewhere, then call `update`/`queue_redraw` and only then draw it in the `_draw` method. With this extension, it would be possible to simplify this to a single method call.
It seemed to me that the addon for this has already been developed for 2D or not?


> In earlier messages you asked to confirm the plugin is installed by searching the help for DebugDraw I believe. In my project and my test repro for this issue, I can't see that in the help.

This was said about Godot 4, where `GDExtension` is used, which tries to load the library as if it were part of the engine itself. Therefore, you can find this class in the documentation.
Godot 3 uses `GDNative`, which simply allows you to use classes and methods from the native library after loading it with `load` and `new`.

**Comment by dogboydog** [comment]
Thanks for your feedback, sorry for my misunderstanding of those methods but i was trying out different settings to see if I could view the lines. 

I see the question about the help is not relevant to the Godot 3.5 version. 
> It seemed to me that the addon for this has already been developed for 2D or not?

Sorry could you rephrase what you mean here? Are you saying it should already work for 2d? Or that there may be some other addon for 2d that I should use instead? In order to use  yours with 2d,  I would need to add a 3d camera to the scene? I don't always run via the godot editor so i wouldn't necessarily be able to use the editor button you show, but maybe I could instantiate a 3d camera via code.  not sure how messy that will get for  a 2d game though

**Comment by DmitriySalnikov** [comment]
> Sorry could you rephrase what you mean here?

I meant that maybe there is another extension for debug drawing in 2d.

> but maybe I could instantiate a 3d camera via code

I'm not sure if there is a easy way to render 3d over 2d. Usually 2d is drawn on top of 3d.

**Comment by dogboydog** [comment]
Ah okay. I haven't found a 2D-enabled one in my searches so far. There is the proposal open to make easier debug drawing an engine feature which seems unlikely to be implemented, which is how I found this addon. 

I found this, which has 2D support but is for Godot 4

https://github.com/jirikostiha/godot-sharp-some

Maybe I should pursue doing it this way for  now:
> in order to draw a line, you need to manually save it somewhere, then call update/queue_redraw and only then draw it in the _draw method.

**Comment by dogboydog** [comment]
Looks like [Line2D](https://docs.godotengine.org/en/3.5/classes/class_line2d.html) could be used for some basic drawing, and you could surround that related could with #if. Thanks for your help finding the problems with my usage of your plugin though, appreciate it

**Comment by hsandt** [comment]
> Usually 2d is drawn on top of 3d.

Indeed, I also tried `DebugDraw3D.draw_line` for a 2D game but it draws the lines behind my 2D scene, whatever Z coord I pass.

For now, I'm using native `draw_line` in `_draw` method, however it uses the current node's Z index and that may make the line show behind the sprite (often the case when sprite is a child of root node with script, at relative Z index 0):

<img width="112" height="93" alt="Image" src="https://github.com/user-attachments/assets/f9a5833c-691c-4306-acfe-95dc5034aa61" />

See those threads:
https://www.reddit.com/r/godot/comments/5vrg0j/how_do_draw_line_above_sprites_set_draw_zvalue/
https://forum.godotengine.org/t/how-to-put-a-draw-line-over-a-sprite/12807
which suggests a few solutions like changing Z index or calling `draw_line` from another node at high Z index, but it's a bit cumbersome. One solution is even to let another node manage drawing all the lines, which is similar to what this add-on does.

So I think this add-on's approach to receiving all drawing requests, deferring them and finally applying them in a centralized place is a good way to solve this.

So completing DebugDraw2D to add methods like `draw_line` would be great to make this work out of the box.
